account/src/main/java/com/user/account/services/UserTransactionService.java

```diff
@@ -1,37 +1,84 @@
 package com.user.account.services;
 
 import java.util.List;
-
 import com.user.account.advice.CustomExceptions;
+import com.user.account.entity.User;
 import com.user.account.message.DefaultMessage;
+import com.user.account.services.AccountService;
+import com.user.account.services.TransactionService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-
 import com.user.account.entity.Transactions;
 import com.user.account.repositories.TransactionRepository;
-
 import javax.transaction.Transactional;
 
-
 @Service
 public class UserTransactionService implements TransactionService {
 
-
     @Autowired
     private TransactionRepository transactionRepository;
 
+    @Autowired
+    private AccountService accountService;
+
+    public ResponseEntity<String> debitImplementation(User user, Long amount){
+        if ( amount.compareTo(user.getAccountBalance()) <= 0){
+            user.decrementAccountBalance(amount);
+            Transactions newTransaction =
+                    new Transactions(user.getAccountNumber(), amount, DefaultMessage.DEBIT);
+            addTransaction(newTransaction);
+            accountService.updateUser(user);
+            return new ResponseEntity<>(DefaultMessage.TRANSACTION_SUCCESS, HttpStatus.OK);
```

Comment:
Just return the success message here, handle sending the `ResponseEntity` with status code in controller itself